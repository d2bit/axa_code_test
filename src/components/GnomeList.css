.frame {
  margin-top: 50px;
  display: flex;
  flex-flow: row wrap;
  padding-bottom: 1rem;
}

.btnBox {
  flex-basis: 100%;
  display: flex;
}

.btn {
  flex-basis: 70%;
  background-color: #4caf50;
  border: 1px solid #8c8c8c;
  color: white;
  font-size: 2rem;
  margin: 1rem auto;
}

@media (min-width: 550px) {
  .frame {
    align-items: stretch;
    align-content: space-around;
  }

  .btn {
    flex-basis: 50%;
    margin: 1rem auto 0;
  }
}

@media (min-width: 850px) {
  .frame {
    align-items: stretch;
    align-content: space-around;
  }

  .btn {
    flex-basis: 30%;
  }
}
