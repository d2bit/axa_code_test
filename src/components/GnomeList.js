import React from 'react'
import Gnome from './Gnome'

function toGnome(profile) {
  return (
    <Gnome key={profile.id} {...profile} />
  )
}
const GnomeList = ({ population }) => {
  return (
    <div>
      { population.map(toGnome) }
    </div>
  )
}

export default GnomeList
