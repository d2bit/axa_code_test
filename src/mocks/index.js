import data from './data.json'

const population = Object.values(data)[0]
const populationCount = population.length

export function getPopulation(size=5) {
  if (size >= populationCount) {
    return population
  }
  return population.slice(0, size)
}

export function getGnome(index) {
  if (index && index >= 0 && index < populationCount) {
    return population[index]
  }
  const randomIndex = Math.floor(Math.random() * populationCount)
  return population[randomIndex]
}

export function getResponseData() {
  return data
}

const Helper = {
  getPopulation,
  getGnome,
  getResponseData,
}

export default Helper
