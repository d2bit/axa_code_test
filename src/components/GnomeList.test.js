import React from 'react'
import { render } from 'react-testing-library'
import { getPopulation } from '../mocks'
import GnomeList from './GnomeList'

describe('GnomeList', () => {
  it('creates Gnomes', () => {
    const populationSize = 3
    const population = getPopulation(populationSize)
    const { container } = render(<GnomeList population={population} />)

    expect(container).toMatchSnapshot()
  })
})
