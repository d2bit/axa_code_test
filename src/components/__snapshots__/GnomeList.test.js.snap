// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GnomeList creates Gnomes 1`] = `
<div>
  <div>
    <div
      data-testid="Gnome-0"
    >
      <h3
        data-testid="name"
      >
        Tobus Quickwhistle
      </h3>
      <img
        alt="Tobus Quickwhistle"
        data-testid="thumbnail"
        src="http://www.publicdomainpictures.net/pictures/10000/nahled/thinking-monkey-11282237747K8xB.jpg"
      />
    </div>
    <div
      data-testid="Gnome-1"
    >
      <h3
        data-testid="name"
      >
        Fizkin Voidbuster
      </h3>
      <img
        alt="Fizkin Voidbuster"
        data-testid="thumbnail"
        src="http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg"
      />
    </div>
    <div
      data-testid="Gnome-2"
    >
      <h3
        data-testid="name"
      >
        Malbin Chromerocket
      </h3>
      <img
        alt="Malbin Chromerocket"
        data-testid="thumbnail"
        src="http://www.publicdomainpictures.net/pictures/30000/nahled/maple-leaves-background.jpg"
      />
    </div>
  </div>
</div>
`;
